TOOLS/odoo-tools.sh

```diff
@@ -1224,23 +1227,72 @@ if [ "$SCRIPT_MODE" = "updatetranslation" ]; then
+    # Todo: check vmware Snapshot how to remote execute the vmware-cmd command with ssh connection to esx server directly, check if the VM is running on this machine
+    # Todo: or find a way of acting through Virtual center server this server has access to the whole cluster and no check on which host a machine is running would be needed
+    # Check if a database with this name exists
+    if [ `su - postgres -c "psql -l | grep ${DBNAME} | wc -l"` -gt 0 ]; then
+        echo -e "Database ${DBNAME} exists, starting to backup this datase ... "
+    elif [ ${DBNAME} = "all" ]; then
+        echo -e "All databases going to be backed up...."
+    elif [ ${DBNAME} = "all_running" ]; then
+        echo -e "All RUNNING databases going to be backed up...."
+    else
+        echo -e "check your Databasename, you gave ${DBNAME}, but this seems not to exist, stopping script......"
+        exit 2
+    fi
+    if [ ${DBNAME} = "all_running" ]; then
+        DATABASE_RUNNING=($(ps -ef|grep "openerp-server*" |awk '{printf $13;printf "\n"; }')) #TODO: check aber auch ALLE Prostgres Prozesse
+        for i in "${DATABASE_RUNNING[@]}"; do
+            #store running databases and log do
+            #getting config of database
+            DATABASECONFIGFILE=${INSTANCE_PATH}/${i}/${i}.conf
+            echo "configflepath --> ${DATABASECONFIGFILE}"
+            BASEPORT69=($(grep "xmlrpc_port" ${DATABASECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+            SUPER_PASSWORD=($(grep "admin_passwd" ${DATABASECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+            BACKUPFILENAME=${INSTANCE_PATH}/${i}/BACKUP/IS-BACKUP--${i}--`date +%Y-%m-%d__%H-%M`.zip
+            echo "backup all Databases, while now backing up ${i} ...."
+            echo -e $(${INSTANCE_PATH}/TOOLS/db-tools.py -b ${BASEPORT69} -s ${SUPER_PASSWORD} "backup" -d ${i} -f ${BACKUPFILENAME}) #-t ${TYPE})
+        done
+    elif [ ${DBNAME} = "all" ]; then
+        DATABASE_RUNNING=($(su - postgres -c "psql --tuples-only -P format=unaligned -c \"SELECT datname FROM pg_database WHERE NOT datistemplate AND datname <> 'postgres'\"|grep -v -e _cloud -e _pad")) #TODO: check aber auch ALLE Prostgres Prozesse
+        for i in "${DATABASE_RUNNING[@]}"; do
+            #store running databases and log do
+            #getting config of database
+            DATABASECONFIGFILE=${INSTANCE_PATH}/${i}/${i}.conf
+            echo "configflepath --> ${DATABASECONFIGFILE}"
+            BASEPORT69=($(grep "xmlrpc_port" ${DATABASECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+            SUPER_PASSWORD=($(grep "admin_passwd" ${DATABASECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+            BACKUPFILENAME=${INSTANCE_PATH}/${i}/BACKUP/IS-BACKUP--${i}--`date +%Y-%m-%d__%H-%M`.zip
+            echo "backup all Databases, while now backing up ${i} ...."
+            echo -e $(${INSTANCE_PATH}/TOOLS/db-tools.py -b ${BASEPORT69} -s ${SUPER_PASSWORD} "backup" -d ${i} -f ${BACKUPFILENAME}) #-t ${TYPE}
```

Comment:
backup gives an error message but runs through the rest of all databases so you see with db is not backed up